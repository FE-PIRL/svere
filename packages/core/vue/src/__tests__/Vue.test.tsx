import { shallowMount } from "@vue/test-utils";

import toVue from "../../src/index";
import SvelteComponent from "../../../../../examples/dist";

describe("Vue", () => {
  let container: any;
  const name = "ben";

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it("should render", async () => {
    const VueComponent = toVue(SvelteComponent);

    const wrapper = shallowMount(VueComponent,{propsData: { name }});

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("count increments when button is clicked", async () => {
    const VueComponent = toVue(SvelteComponent);

    const wrapper = shallowMount(VueComponent,{propsData: { name }});
    expect(wrapper.find("button").text()).toBe("count: 0");

    await wrapper.find("button").trigger("click");

    expect(wrapper.find("button").text()).toBe("count: 1");
  });

  it("watch internal props changes and execute the callback", async () => {
    const VueComponent = toVue(SvelteComponent);

    const handleSomeEvent = (name: string) => {
      //console.log(name)
      expect(name).toBe("yasin");
    };
    const wrapper = shallowMount(VueComponent, {
      propsData: { name },
      listeners: {
        "watch:name": handleSomeEvent,
      },
    });

    await wrapper.find("button").trigger("click");
  });

  it("listen custom event", async () => {
    const VueComponent = toVue(SvelteComponent);

    const handleSomeEvent = (e: any) => {
      //console.log(e.detail);
      expect(e.detail).toBe(1);
    };

    const wrapper = shallowMount(VueComponent, {
      propsData: { name },
      listeners: {
        someEvent: handleSomeEvent,
      },
    });

    await wrapper.find("button").trigger("click");
  });

  it("custom wrapper", async () => {
    const wrapperProps = {
      element: "section",
      className: "section-css",
      id: "svelte-vue2",
      styles: {
        border: "1px solid gray",
      },
    };
    const VueComponent = toVue(SvelteComponent, wrapperProps);

    const wrapper = shallowMount(VueComponent,{propsData: { name }});

    expect(wrapper.html()).toMatchSnapshot();
  });
});
