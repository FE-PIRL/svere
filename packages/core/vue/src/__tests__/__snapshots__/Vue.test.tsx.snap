// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vue custom wrapper 1`] = `
"<section id=\\"svelte-vue2\\" class=\\"section-css\\" style=\\"border: 1px solid gray;\\" name=\\"ben\\">
  <main class=\\"svelte-1tky8bj\\">
    <h1 class=\\"svelte-1tky8bj\\">Hello ben!</h1> <button>count: 0</button>
  </main>
</section>"
`;

exports[`Vue should render 1`] = `
"<div id=\\"svelte-wrapper\\" name=\\"ben\\">
  <main class=\\"svelte-1tky8bj\\">
    <h1 class=\\"svelte-1tky8bj\\">Hello ben!</h1> <button>count: 0</button>
  </main>
</div>"
`;
